tests/main/interfaces-classic-content-slot/task.yaml

```diff
@@ -0,0 +1,60 @@
+summary: Ensure that the content slots for classic snaps works.
+
+details: |
+    The content-sharing interface interface allows a snap to access contents from
+    other snap. It is possible to have content slots in classic snaps, where
+    confined snaps can connect to.
+
+    A snap which defines the content sharing plug must be shown in the interfaces list.
+    The plug must be autoconnected on install and, as usual, must be able to be
+    reconnected.
+
+# slow in autopkgtest (>4m)
+backends: [-autopkgtest]
+
+# We use classic snaps for the test, so it cannot run on core
+systems: [-ubuntu-core-*]
+
+prepare: |
+    echo "Ensure an empty state so that installing test-snapd-content-plug-to-classic-slot"
+    echo "will pull in test-snapd-classic-content-slot *and* core"
+
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB/systemd.sh"
+
+    systemd_stop_units snapd.service
+    rm -f /var/lib/snapd/state.json
+    systemctl start snapd
+    snap wait system seed.loaded
+
+execute: |
+    echo "When a snap declaring a content sharing plug is installed"
+    snap install --edge test-snapd-content-plug-to-classic-slot
```

Comment:
In case you declare the snap locally, you can install it by doing
"$TESTSTOOLS"/snaps-state install-local test-snapd-content-plug-to-classic-slot
This will automatically remove during restore